app/src/main/java/com/infomaniak/drive/ui/menu/GalleryViewModel.kt

```diff
@@ -17,69 +17,87 @@
  */
 package com.infomaniak.drive.ui.menu
 
-import androidx.lifecycle.*
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
 import com.infomaniak.drive.data.api.ApiRepository
+import com.infomaniak.drive.data.api.CursorApiResponse
 import com.infomaniak.drive.data.cache.FileController
 import com.infomaniak.drive.data.models.File
 import com.infomaniak.drive.utils.IsComplete
-import com.infomaniak.drive.utils.isLastPage
-import com.infomaniak.lib.core.models.ApiResponse
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.Job
+import kotlinx.coroutines.launch
 
 class GalleryViewModel : ViewModel() {
-    private var getGalleryJob: Job = Job()
 
-    var lastGalleryPage = 1
-    var lastGalleryLastPage = 1
+    private var getGalleryJob: Job? = null
 
-    val loadMoreGallery = MutableLiveData<Pair<Int, Boolean>>()
-    val galleryApiResult = loadMoreGallery.switchMap { (driveId, ignoreCloud) ->
-        getLastGallery(driveId, ignoreCloud)
+    private var currentCursor: String? = null
+
+    val galleryApiResult = MutableLiveData<Pair<ArrayList<File>, IsComplete>?>()
+
+    fun loadLastGallery(driveId: Int, ignoreCloud: Boolean) {
+        getGalleryJob?.cancel()
+        getGalleryJob = viewModelScope.launch(Dispatchers.IO) {
+            galleryApiResult.postValue(getLastGallery(driveId, ignoreCloud, isFirstPage = true))
+        }
+    }
+
+    fun loadMoreGallery(driveId: Int, ignoreCloud: Boolean) {
+        currentCursor?.let {
+            getGalleryJob?.cancel()
+            getGalleryJob = viewModelScope.launch(Dispatchers.IO) {
+                galleryApiResult.postValue(getLastGallery(driveId, ignoreCloud, isFirstPage = false, currentCursor))
+            }
+        }
     }
 
     private fun getLastGallery(
         driveId: Int,
         ignoreCloud: Boolean = false,
-    ): LiveData<Pair<ArrayList<File>, IsComplete>?> {
-        getGalleryJob.cancel()
+        isFirstPage: Boolean = true,
```

Comment:
The default value of isFirstPage is useless as it's 1 false and 1 true
Either remove this or the `isFirstPage = true`  in `loadLastGallery`